motionplan/planManager.go

```diff
@@ -612,6 +634,60 @@ func (pm *planManager) goodPlan(pr *rrtPlanReturn, opt *plannerOptions) (bool, f
 	return false, solutionCost
 }
 
+func (pm *planManager) planToRRTGoalMap(plan Plan, goal spatialmath.Pose) (*rrtMaps, error) {
+	planNodes := make([]node, 0, len(plan))
+	// Build a list of nodes from the plan
+	for _, planStep := range plan {
+		conf, err := pm.frame.mapToSlice(planStep)
+		if err != nil {
+			return nil, err
+		}
+		planNodes = append(planNodes, newConfigurationNode(conf))
+	}
+
+	if pm.useTPspace {
+		// Fill in positions from the old origin to where the goal was during the last run
+		planNodesOld, err := rectifyTPspacePath(planNodes, pm.frame, spatialmath.NewZeroPose())
+		if err != nil {
+			return nil, err
+		}
+
+		// Figure out where our new starting point is relative to our last one, and re-rectify using the new adjusted location
+		oldGoal := planNodesOld[len(planNodesOld)-1].Pose()
+		pathDiff := spatialmath.PoseBetween(oldGoal, goal)
+		planNodes, err = rectifyTPspacePath(planNodes, pm.frame, pathDiff)
+		if err != nil {
+			return nil, err
+		}
+	}
+	// Fill in costs
+	for i := 1; i < len(planNodes); i++ {
```

Comment:
I'm wondering if this means we should be extending the definition of plans.  I'm imagining we return as the raw rrt map data and this struct has only one public method to extract the plan, so that effectively it looks the same from a users perspective but we can do more complicated things in the motionplan package that would make redundant work like this unnecessary

Reply:
We could explore that as part of [RSDK-2503](https://viam.atlassian.net/browse/RSDK-2503)